src/java/org/apache/cassandra/io/util/CompressedChunkReader.java

```diff
@@ -117,8 +141,23 @@ public void readChunk(long position, ByteBuffer uncompressed)
                 {
                     ByteBuffer compressed = bufferHolder.getBuffer(length);
 
-                    if (channel.read(compressed, chunk.offset) != length)
-                        throw new CorruptBlockException(channel.filePath(), chunk);
+                    if (readAheadBuffer != null && readAheadBuffer.hasBuffer())
```

Comment:
Why do you only handle this block?  There are 2 blocks that logically would want to do this

Reply:
@dcapwell apologies, not sure I totally grok this one either. Do you mean why do we only read one block at a time?